api/provisioner/machine.go

```diff
@@ -227,6 +227,27 @@ func (m *Machine) Series() (string, error) {
 	return result.Result, nil
 }
 
+// AvailabilityZone returns an underlying provider's availability zone
+// for a machine
+func (m *Machine) AvailabilityZone() (string, error) {
```

Comment:
This needs a test.

Reply:
https://github.com/juju/juju/pull/7795/commits/fb986bbbad2c377610760f883c76297eab2655fb